src/shiv/bootstrap/__init__.py

```diff
@@ -99,11 +105,14 @@ def bootstrap():
     if not site_packages.exists() or env.force_extract:
         extract_site_packages(archive, site_packages.parent)
 
-    preserved = sys.path[1:]
+    # Find the first instance of an existing site-packages on sys.path
+    index = _first_sitedir_index() or len(sys.path)
+
+    # preserve the tail
+    preserved = sys.path[index:]
 
-    # truncate the sys.path so our package will be at the start,
-    # and take precedence over anything else (eg: dist-packages)
-    sys.path = sys.path[0:1]
+    # truncate sys.path in preparation for extension
+    sys.path = sys.path[:index]
```

Comment:
Truncating `sys.path` while making the function calls, such as `site.addsitedir()` below may not be the best approach. Why not make a call first, take that tail off (that no one is using yet), and insert it at index after that. Feels safer than removing a part that's potentially already in use.